From: Jiri Slaby <jslaby@suse.cz>
Date: Thu, 25 Jan 2018 14:17:07 +0100
Subject: Revert "futex: Prevent overflow by strengthen input validation"
Patch-mainline: submitted 25/1/2018
References: 4.14.15-fix

This reverts commit 17ae6ccfe5dd85605dc44534348b506f95d16a61, upstream
commit fbe0e839d1e22d88810f3ee3e2f1479be4c0aa4a. It breaks strace's
tests:
futex(0x7ff568b44ffc, 0x3, 0xfacefeed, 0xbadda7a0ca7b100d, 0x7ff568b44ffc, 0x9caffee1) = -1: Invalid argument

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 kernel/futex.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/kernel/futex.c b/kernel/futex.c
index 52b3f47031..29ac5b64e7 100644
--- a/kernel/futex.c
+++ b/kernel/futex.c
@@ -1878,9 +1878,6 @@ static int futex_requeue(u32 __user *uaddr1, unsigned int flags,
 	struct futex_q *this, *next;
 	DEFINE_WAKE_Q(wake_q);
 
-	if (nr_wake < 0 || nr_requeue < 0)
-		return -EINVAL;
-
 	/*
 	 * When PI not supported: return -ENOSYS if requeue_pi is true,
 	 * consequently the compiler knows requeue_pi is always false past
-- 
2.15.1

